adsft/extraction.py

```diff
@@ -587,14 +593,20 @@ def extract_multi_content(self, translate=False, decode=False):
                         translate=translate,
                     )
 
-
                     if text_content:
-                        meta_out[content_name] = text_content
+                        # ensure content is not repeated due to nested xpaths

```

Comment:
You found nested xpaths? How are we sure we are grabbing the good one? I am afraid of situation like this:

```
<xpath1>
  text1
  <xpath2>
    text2
  </xpath2>
</xpath1>
```

If we want text1 + text2, we would like to grab xpath1 and ignore xpath2. That would be the case if our list of xpaths is [xpath1, xpath2], but if it is [xpath2, xpath1] or the full text has these tags reverted, then we would concatenate text2 and text1 + text2.

Reply:
by nested xpaths I was referring to the appendix tag being found inside of the body tag in some xml files, when this happens and how often I have not yet determined. We only need to worry about the above situation occurring if the rules.py file is modified so that the appendix tag is not ordered after the body tags. Comments have been added to explain this. 

Another solution would be to test that content from the previous tag(s) is not in the new tag in addition to the current method of testing that content from the current tag is not in any of the previously found content. We decided against this solution for now as it would most likely decrease speed significantly.